### PR TITLE
feat: tables, alignment, anchor tag, title and interface improvements

### PR DESCRIPTION
**Pull request by rs-amp**
This PR adds some additional tools to the editor, and changes some existing ones:

- Soft Hyphen (`&shy;`)
  - Node that draws as a semi-transparent dash.
- Anchor Tag (`<a id="anchor-id"></a>`)
  - Note that draws as an anchor in a box with its id shown when hovering.
- Alignment (when not left aligned, `<p style="text-align:right;">...</p>`)
  - Alignment within tables or other blocks is also done this way.
- Tables
  - Cells cannot be merged. The first row of the table will always be the table header. You should expect the same limitations as markdown tables, and set up your styles accordingly.
---
- Links now show their previous title and url when you attempt to create a link in the same position, making them easier to edit.
- Inline Styles can now be used in the JSON output mode.
- A document has been included that demonstrates different kinds of tool, and what their output result looks like.

## Implementation
The new tools have been implemented in the `language-markdown` project, as some of them piggyback on the same html_inline handling as inline styles (anchor, hard break in tables) which are now initialized as part of the json language as well. All other tools require custom markdown parsers and serializers, so have been added to the markdown language since most of the tool work is done getting them working in markdown anyways.

For the consuming side, markdown renderers used _must_ allow inline html if they want to use:
  - Anchor (`a[id]`)
  - \<br> in tables
  - Inline Styles (`span[class]`)
  - Alignment (`p[align]`)
You could potentially extend your markdown parser to filter out just the tags we use if you don't want other tags being used.

I'm not sure I like the toolbar getting as long as it is. Maybe we should come up with another way to inject or remove toolbar items from the default, rather than making users redefine the whole thing.

### CODE REVIEW

**Comment by rs-amp** [comment]
A few additional changes have been made:
- There is now a `title` parameter, which displays a title above the rich text editor when present.
- Since raw HTML tags are used for a few features, replace any reserved HTML characters like `<` with escaped variants like `&lt;`. This avoids some confusing errors.
- Alignment is now done via a `style` attribute (with text-align), or a `class` attribute of predefined name.
  - The inline style method produces code like `<p style="text-align:right;">...</p>`
  - The class method uses predefined classes. It is activated by passing `useClasses` as true in the initialisation parameters.
    - The classes are, by default, in the format `amp-align-<alignment>`, eg. `amp-align-left`. You can override these class names with a `classOverride` parameter: `"classOverride": {"amp-align-left": "custom-align-right"}
- When creating an anchor, it now enforces a globally unique anchor ID, and informs you when the provided ID is not valid.
- Changed some key behaviour with lists to better match Word, or other wysiwyg editors. It's not a perfect match, as our structure for lists is a bit different.

**Review by PikinerLiliya — APPROVED** [review]
Great job

File: HOSTING.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Hosting the Rich Text Extension
```
Comment (easen-amp) [inline HOSTING.md:1]: We have a change to host this extension rich-text.extensions.content.amplience.net, which will be released soon. Might be best to hold off merging this down until his hosting is in place.

**Review by easen-amp — COMMENTED** [review]
We have a change to host this extension rich-text.extensions.content.amplience.net, which will be released soon. Might be best to hold off merging this down until his hosting is in place.

**Comment by rs-amp** [comment]
Updated HOSTING.md to reflect the new hosting arrangements.

**Review by easen-amp — CHANGES_REQUESTED** [review]
Can the PR be renamed to start with "feat: " and include a sensible feature name as this will be recorded in the CHANGELOG.md when merge down.

**Comment by rs-amp** [comment]
> Can the PR be renamed to start with "feat: " and include a sensible feature name as this will be recorded in the CHANGELOG.md when merge down.

Is the new name alright?

**Review by easen-amp — APPROVED** [review]

**Pull request merged**